model/src/gateway/payload/typing_start.rs

```diff
@@ -1,14 +1,217 @@
 use crate::{
-    guild::Member,
+    guild::member::{Member, MemberDeserializer},
     id::{ChannelId, GuildId, UserId},
 };
-use serde::{Deserialize, Serialize};
+use serde::{
+    de::{DeserializeSeed, Deserializer, Error as DeError, MapAccess, Visitor},
+    Deserialize, Serialize,
+};
+use serde_value::Value;
+use std::fmt::{Formatter, Result as FmtResult};
 
-#[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
+#[derive(Clone, Debug, Eq, Hash, PartialEq, Serialize)]
 pub struct TypingStart {
     pub channel_id: ChannelId,
     pub guild_id: Option<GuildId>,
+    pub member: Option<Member>,
     pub timestamp: u64,
     pub user_id: UserId,
-    pub member: Option<Member>,
+}
+
+#[derive(Debug, Deserialize)]
+#[serde(field_identifier, rename_all = "snake_case")]
+enum Field {
+    ChannelId,
+    GuildId,
+    Member,
+    Timestamp,
+    UserId,
+}
+
+struct TypingStartVisitor;
+
+impl<'de> Visitor<'de> for TypingStartVisitor {
+    type Value = TypingStart;
+
+    fn expecting(&self, f: &mut Formatter<'_>) -> FmtResult {
+        f.write_str("struct TypingStart")
+    }
+
+    fn visit_map<V: MapAccess<'de>>(self, mut map: V) -> Result<Self::Value, V::Error> {
+        let mut channel_id = None;
+        let mut guild_id = None::<Option<_>>;
+        let mut member = None::<Option<Value>>;
+        let mut timestamp = None;
+        let mut user_id = None;
+
+        loop {
+            let key = match map.next_key() {
+                Ok(Some(key)) => key,
+                Ok(None) => break,
+                Err(_) => {
+                    // Encountered when we run into an unknown key.
+                    continue;
```

Comment:
hmm, do we want to log if we come up on a unknown key?

Reply:
I was thinking this issue would be a good solution for that https://github.com/twilight-rs/twilight/issues/44

Reply:
But that doesn't cover the use cases outside of the gateway and http, so maybe we do want to?

Reply:
I honestly think we should log whenever we come across unknown keys, I'd be in favor of updating #44 to include to model as well.

Reply:
Updated